api/src/services/application-csv-export.service.ts

```diff
@@ -815,15 +699,389 @@ export class ApplicationCsvExporterService
+   * @param queryParams the incoming param args
+   * @param forLottery whether we are getting the lottery results or not
+   * @returns void but writes the output to a file
+   */
+  csvExportHelper(
+    filename: string,
+    applications: Application[],
+    csvHeaders: CsvHeader[],
+    queryParams: ApplicationCsvQueryParams,
+    forLottery = false,
+  ): Promise<void> {
+    return new Promise(async (resolve, reject) => {
+      // create stream
+      const writableStream = fs.createWriteStream(`${filename}`);
+      writableStream
+        .on('error', (err) => {
+          console.log('csv writestream error');
+          console.log(err);
+          reject(err);
+        })
+        .on('close', () => {
+          resolve();
+        })
+        .on('open', async () => {
+          writableStream.write(
+            csvHeaders
+              .map((header) => `"${header.label.replace(/"/g, `""`)}"`)
+              .join(',') + '\n',
+          );
+
+          const promiseArray: Promise<string>[] = [];
+          for (let i = 0; i < applications.length; i += NUMBER_TO_PAGINATE_BY) {
+            promiseArray.push(
+              new Promise(async (resolve) => {
+                // grab applications NUMBER_TO_PAGINATE_BY at a time
+                let paginatedApplications =
+                  await this.prisma.applications.findMany({
+                    include: {
+                      ...view.csv,
+                      demographics: queryParams.includeDemographics
+                        ? {
+                            select: {
+                              id: true,
+                              createdAt: true,
+                              updatedAt: true,
+                              ethnicity: true,
+                              gender: true,
+                              sexualOrientation: true,
+                              howDidYouHear: true,
+                              race: true,
+                            },
+                          }
+                        : false,
+                      applicationLotteryPositions: forLottery
+                        ? {
+                            select: {
+                              ordinal: true,
+                            },
+                            where: {
+                              multiselectQuestionId: null,
+                            },
+                          }
+                        : false,
+                    },
+                    where: {
+                      listingId: queryParams.listingId,
+                      deletedAt: null,
+                    },
+                    skip: i,
+                    take: NUMBER_TO_PAGINATE_BY,
+                  });
+                if (forLottery) {
+                  paginatedApplications = paginatedApplications.sort(
+                    (a, b) =>
+                      a.applicationLotteryPositions[0].ordinal -
+                      b.applicationLotteryPositions[0].ordinal,
```

Comment:
When I ran this with production data the export failed on this line because "ordinal" didn't exist on some applications. And looking at the exported data there are several chunks where the rank doesn't show up 
![image](https://github.com/user-attachments/assets/0c87cf83-5189-4b39-90c3-855b2e1f2341)
I'll keep investigating on why this might be.

Reply:
Hey @ludtkemorgan I think I figured it out, chunks where the rank is missing is probably tied with those being marked as duplicate which shouldn't show in the export 

I made an adjustment that I'll be pushing up that should remove duplicates from the lottery export completely and solve both the ordinal is missing problem in the sort and removes this unranked chunks from the export

Reply:
The confirmation string here might also need a fix - I think it shows correctly for an applicant, so if this was the confirmation string used in a lottery, it wouldn't match.

Reply:
We also want duplicates to be in the lottery / allow for people to do that

Reply:
This issue no longer exists, but I think the sorting is not properly working. It appears to maybe just be sorting the applications in the chunks that it works on. So it sorts 500 at a time. 
![image](https://github.com/user-attachments/assets/4991a7e8-3a63-46f2-9f83-63ea14a815d0)